_posts/2020-01-30-Age-Targeting-Part1.md

```diff
@@ -0,0 +1,129 @@
+![example_age_distribution](../images/Age-Targeting/example_age_distribution.png)
+Different apps clearly show different age distributions..
+However, it is clear that some of the distributions in the plot do not make sense.
+Why would people from 17-18 correspond to more than 95% of the total users of the app? (What happens to the 16ties? or 19ties?)
+Why would apps have the same numbers of devices for every age in a specific range?
+These must be very special apps, or the data do not correspond to truth.
+What we want to do is to exclude these apps.
+Since we want to make a clear statement about the age distribution of the devices which visit a certain app, we exclude those where the amount of users is not enough to draw any statistical conclusion.
+This is not a big issue since the devices contained in these apps are few by definition, so we do not cut off too many data in this way.
+
+Let's describe the main pattern we discover in the data and how we get rid of them.
+
+* **Few ages**<br/>
+Certain apps seem to provide almost always only two/three values.
+We create a variable (frac12) which is the fraction of devices contained in the two top most frequent ages observed in the app.
+Apps where the frac12 value is higher than 30% are excluded.
+Below you can see the distribution of the frac12 variable and the consequences of the rejection of the non reliable apps on the total age distribution.
+![frac12_cut](../images/Age-Targeting/frac12_cut.png)
+The red vertical line indicates the threshold we apply over the frac12 feature.
+
+
+* **Uniform distribution**<br/>
+One of the four examples (the bottom left one) clearly show a uniform generated distribution.
+Detect those apps is relatively easy, since for a fixed device a different age will be provided every time the device accesses the app.
+For every devices which have been observed at least twice in the app, we calcualte the max - min age value and, then, we calculate the average of this differences.
+In the image you can see the distribution of the avg_range_age variable and the effect of the cut on the global distributions.
+![frac12_cut](../images/Age-Targeting/uniform_cut.png)
+
+
+* **Symmetric like shapes**<br/>
+Certain distributions show a shape which seems to be suspicious symmetric.
+Using the [skewness](https://en.wikipedia.org/wiki/Skewness) statistical measure we can select those and drop them out.
+In the plot below you can see the distribution of the skewness and the effect of the cut on the global distributions.
+![frac12_cut](../images/Age-Targeting/skewness_cut.png)
+
+
+* **Too many peaks**<br/>
+Using a basic peak detecting algorithm, written in pyspark and reported below, we can count the number of peaks on the age distribution.
+```python
+from pyspark.sql import functions as F
+from pyspark.sql.types import IntegerType
+from pyspark.sql import Window
+def get_apps_with_peak_information(df):
```

Comment:
newline in here

Reply:
the python cell won't be complete anymore if I add a newline here, I am doing something wrong in markdown syntax?